WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift

```diff
@@ -358,22 +358,8 @@ class AztecPostViewController: UIViewController, PostEditor {
 
     /// Maintainer of state for editor - like for post button
     ///
-    fileprivate lazy var postEditorStateContext: PostEditorStateContext! = {
-        var originalPostStatus: BasePost.Status? = nil
-
-        if let originalPost = self.post.original,
-            let postStatus = originalPost.status,
-            originalPost.hasRemote() {
-            originalPostStatus = postStatus
-        }
-
-
-        // Self-hosted non-Jetpack blogs have no capabilities, so we'll default
-        // to showing Publish Now instead of Submit for Review.
-        let userCanPublish = self.post.blog.capabilities != nil ? self.post.blog.isPublishingPostsAllowed() : true
-        let context = PostEditorStateContext(originalPostStatus: originalPostStatus, userCanPublish: userCanPublish, publishDate: self.post.dateCreated, delegate: self)
-
-        return context
+    fileprivate lazy var postEditorStateContext: PostEditorStateContext = { [unowned self] in
```

Comment:
I don't think we need the `[unowned self]` here? (https://twitter.com/jckarter/status/704100315587477504)

Reply:
But you're not immediately applying the closure (even if the syntax makes it look that way). Since it's a `lazy` variable, it will only initialize when it's first used.

Reply:
(What Koke said) =D. Thanks Nate!

Reply:
Does https://twitter.com/jckarter/status/704100315587477504 help?

Reply:
@jleandroperez - I tested a few days ago and @nheagy 's comment is right.  It seems like these lazy-var initialization blocks are smart enough not to retain self at all (probably by having `[unowned self]` by default).

I basically set up a playground and tried using the lazy var, and never using it, and in both cases the object owning the variable would be released before execution ended (`deinit` was called).